droidlet/dashboard/web/src/components/Memory2D.js

```diff
@@ -251,87 +239,70 @@ class Memory2D extends React.Component {
     if (obstacle_map) {
       obstacle_map.forEach((obj) => {
         let color = "#827f7f";
-        let x = parseInt(((obj[0] - xmin) / (xmax - xmin)) * width);
-        let y = parseInt(((obj[1] - ymin) / (ymax - ymin)) * height);
-        renderedObjects.push(
-          // FIXME: may need to also add mobile support? by using onTap/onTouchEnd
-          <Circle
-            key={j++}
-            radius={2}
-            x={x}
-            y={y}
-            fill={color}
-            onClick={(e) => {
-              this.handleObjClick("obstacle_map", x, y, {
-                memid: "don't edit",
-                x: obj[0],
-                y: obj[1],
-              });
-            }}
-            onMouseOver={(e) => {
-              e.currentTarget.setRadius(5);
-            }}
-            onMouseOut={(e) => {
-              e.currentTarget.setRadius(2);
-            }}
-          />
-        );
+        let map_x = parseInt(((obj[0] - xmin) / (xmax - xmin)) * width);
+        let map_y = parseInt(((obj[1] - ymin) / (ymax - ymin)) * height);
+        let map_pos = "" + map_x + "," + map_y;
+        let poolData = {
+          type: "obstacle_map",
+          radius: 2,
+          radiusFocused: 5,
+          color: color,
+          data: {
+            memid: "don't edit" + j++,
+            x: obj[0],
+            y: obj[1],
+            pos: "[" + obj[0] + ",0," + obj[1] + "]",
+          },
+        };
+        if (!(map_pos in objectPosPool)) {
+          objectPosPool[map_pos] = [poolData];
+        } else {
+          objectPosPool[map_pos].push(poolData);
+        }
       });
     }
 
-    // Put detected objects from memory on map
+    // Pool detected objects from memory by position
     detections_from_memory.forEach((obj) => {
-      let obj_id = obj.obj_id;
       let xyz = obj.pos;
       let color = "#0000FF";
-      let x = parseInt(((xyz[2] - xmin) / (xmax - xmin)) * width);
-      let y = parseInt(((-xyz[0] - ymin) / (ymax - ymin)) * height);
-      y = height - y;
-      renderedObjects.push(
-        <Circle
-          key={obj.memid}
-          radius={3}
-          x={x}
-          y={y}
-          fill={color}
-          onClick={(e) => {
-            this.handleObjClick("detection_from_memory", x, y, obj);
-          }}
-          onMouseOver={(e) => {
-            e.currentTarget.setRadius(6);
-          }}
-          onMouseOut={(e) => {
-            e.currentTarget.setRadius(3);
-          }}
-        />
-      );
+      let [map_x, map_y] = this.convertCoordinate(xyz);
+      let map_pos = "" + map_x + "," + map_y;
+      let poolData = {
+        type: "detection_from_memory",
+        radius: 6,
+        radiusFocused: 9,
+        color: color,
+        data: obj,
+      };
+      if (!(map_pos in objectPosPool)) {
+        objectPosPool[map_pos] = [poolData];
+      } else {
+        objectPosPool[map_pos].push(poolData);
+      }
     });
 
+    // Put objects from undefined memory on map?
```

Comment:
This block doesn't have any purpose with CraftAssist? Keeping in case it's used for Robot